File: Listribute/app/model/item.ts
export class Item {
    readonly id: number;
    readonly name: string;
    readonly description?: string | null;
    readonly imgUrl?: string | null;
    readonly listId: number;
    readonly order: number;
    private _checkedBy: string[];
    get checkedBy() {
        return this._checkedBy.slice();
    }
    set checkedBy(value: string[]) {
        this._checkedBy = value.slice();
    }

    constructor(from: any) {
        // TODO: Input validation
        this.id = from.id;
        this.name = from.name;
        this.description = from.description;
        this.imgUrl = from.imgUrl;
        this.listId = from.listId;
        this.order = from.order;
        this._checkedBy = from.checkedBy?.slice() ?? [];
    }
}

File: Listribute.Core/Model/Item.cs
using System;
using System.Collections.Generic;

namespace Listribute.Core.Model
{
    public class Item
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImgUrl { get; set; }
        public List List { get; set; }
        public int Order { get; set; }
        public List<User> CheckedBy { get; set; }
        public DateTimeOffset Created { get; set; }
        public User CreatedBy { get; set; }

        public Item(
            string name,
            string description,
            string imgUrl,
            List list,
            int order,
            List<User> checkedBy,
            DateTimeOffset created,
            User createdBy)
        {
            Name = name;
            Description = description;
            ImgUrl = imgUrl;
            List = list;
            Order = order;
            CheckedBy = checkedBy;
            Created = created;
            CreatedBy = createdBy;
        }
    }
}
